### PR TITLE
chore: use azure containerd packages

### LINKED ISSUE #2370: containerd integration using deprecated "cri" binary
**Describe the bug**

For containerd we are fetching the deprecated standalone cri-containerd binary.
CRI is now compiled into containerd directly, this should be used instead.

**Additional context**

https://github.com/Azure/aks-engine/blob/f8f373d241780a7945e40cdc9dc0d4b326d334b2/parts/k8s/cloud-init/artifacts/cse_install.sh#L173-L178

From cri-containerd repo:

> Note: The standalone cri-containerd binary is end-of-life. cri-containerd is transitioning from a standalone binary that talks to containerd to a plugin within containerd. This github branch is for the cri plugin. See standalone-cri-containerd branch for information about the standalone version of cri-containerd.

I already have some packaging for containerd, but have not tested extensively or finalized the pipeline to publish such packages.

### PR DESCRIPTION
**Pull request by cpuguy83**
**Reason for Change**:

**Issue Fixed**:

Fixes #2370

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

This is a POC.
Looking for feedback on the implementation.
Also need tests to run, currently having issues deploying things.

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_install.sh [hunk 1]
```diff
@@ -227,18 +231,14 @@ installContainerd() {
...
-        rm -Rf /usr/bin/containerd
-        rm -Rf /var/lib/docker/containerd
-        rm -Rf /run/docker/containerd
-        if [[ ! -f "$CONTAINERD_DOWNLOADS_DIR/${CONTAINERD_TGZ_TMP}" ]]; then
-            downloadContainerd
-        fi
-        tar -xzf "$CONTAINERD_DOWNLOADS_DIR/$CONTAINERD_TGZ_TMP" -C /
-        sed -i '/\[Service\]/a ExecStartPost=\/sbin\/iptables -P FORWARD ACCEPT -w' /etc/systemd/system/containerd.service
-        echo "Successfully installed cri-containerd..."
+        removeContainerd
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_install.sh]: Shall I infer that this is a "best practice, just-in-case" type operation? In other words, the most deterministic way to install a particular version of the `moby-containerd` is to ensure no pre-existing `moby-containerd` and `moby-runc` packages aren't present on the OS already?

Also, is `moby-runc` installed as a dependency of `moby-containerd`? Or is it just a conflicting package and so we need to ensure it's removed before we install `moby-containerd`?

Comment (cpuguy83) [reply parts/k8s/cloud-init/artifacts/cse_install.sh]: I copied this pattern from `installMoby` is all.

`moby-runc` is a dependency for `moby-containerd` and is automatically installed with `moby-containerd`.

**Comment by jackfrancis** [comment]
@cpuguy83 Are the `moby-engine` and `moby-cli` installed packages able to co-exist with `moby-containerd`? If so, might we consider pre-installing `moby-containerd` on the VHD so we don't have to fetch all the apt artifacts during bootstrap runtime?

**Comment by cpuguy83** [comment]
> Are the moby-engine and moby-cli installed packages able to co-exist with moby-containerd

`moby-cli` yes, `moby-engine`, not yet...
`moby-containerd` explicitly lists `moby-engine` as a conflict right now.

File: parts/k8s/cloud-init/artifacts/cse_install.sh [hunk 2]
```diff
@@ -227,18 +231,14 @@ installContainerd() {
...
+        echo "deb [arch=amd64,arm64,armhf] https://packages.microsoft.com/ubuntu/${UBUNTU_RELEASE}/multiarch/prod testing main" > /tmp/microsoft-prod-testing.list
+        retrycmd_if_failure 10 5 10 cp /tmp/microsoft-prod.list /etc/apt/sources.list.d/ || exit $ERR_MOBY_APT_LIST_TIMEOUT
```
Comment (cpuguy83) [inline parts/k8s/cloud-init/artifacts/cse_install.sh]: Oops

File: parts/k8s/cloud-init/artifacts/cse_install.sh (hunk 1 above)
Comment (jackfrancis) [reply parts/k8s/cloud-init/artifacts/cse_install.sh]: Got it.

Should we invoke `removeMoby` at the beginning of the containerd installation process to ensure we don't ship both packages?

Likewise we should probably invoke `removeContainerd` during moby installation.

**Pull request merged**